src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,206 @@
+      border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    }
 
-    // Unchecked toggle switch
-    ~ .iui-toggle {
-      display: block;
-      position: relative;
-      height: $iui-baseline * 2 - $iui-xxs;
-      width: ($iui-m * 2) + ($iui-xxs * 3);
-      border-radius: $iui-baseline;
+    &::before {
+      content: '';
+      position: absolute;
+      inset: 0;
+      transition: border-color $iui-speed-fast ease-out;
+      border-radius: inherit;
+      border-style: solid;
+      border-width: 1px;
+      border-color: inherit;
+      box-sizing: border-box;
+    }
+
+    &::after {
+      content: '';
+      position: absolute;
+      height: $toggle-switch-height - ($iui-xxs * 3);
+      width: $toggle-switch-height - ($iui-xxs * 3);
+      aspect-ratio: 1 / 1;
+      top: 50%;
+      transform: translateY(-50%);
+      border-radius: 50%;
       @media (prefers-reduced-motion: no-preference) {
-        transition: background-color $iui-speed-fast ease, border-color $iui-speed-fast ease;
+        transition: left $iui-speed-fast ease, background-color $iui-speed-fast ease, opacity $iui-speed-fast ease;
       }
+    }
+
+    // Unchecked toggle switch
+    &:not(:checked) {
       @include themed {
         background-color: t(iui-color-background-1);
-        border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+        border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
       }
 
-      > .iui-icon {
-        opacity: 0;
-        width: $iui-sm;
-        height: $iui-sm;
-        display: inline-block;
-        position: relative;
-        left: $iui-xs;
-        @media (prefers-reduced-motion: no-preference) {
-          transition: opacity $iui-speed-fast ease;
-        }
-        @include themed {
-          fill: t(iui-color-foreground-accessory);
-        }
-      }
-
-      > .iui-handle {
-        position: absolute;
-        height: $iui-m;
-        width: $iui-m;
-        top: 50%;
-        transform: translateY(-50%);
-        right: $iui-m + $iui-xs;
-        border-radius: 50%;
-        @media (prefers-reduced-motion: no-preference) {
-          transition: right $iui-speed-fast ease, background-color $iui-speed-fast ease, opacity $iui-speed-fast ease;
-        }
+      &::after {
+        left: 3px;
         @include themed {
           background-color: t(iui-color-foreground-body);
           opacity: t(iui-opacity-2);
         }
       }
-    }
 
-    &:hover:not(:disabled),
-    &:focus {
-      ~ .iui-toggle {
+      &:hover:enabled,
+      &:focus {
         @include themed {
-          border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
+          border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
         }
+      }
 
-        > .iui-handle {
-          @include themed {
-            opacity: t(iui-opacity-1);
-          }
+      &:focus:not(:focus-visible) {
```

Comment:
Can you use our `focus-visible` mixin here?

Reply:
I am using the `focus-visible` mixin on lines 126 & 162.

Reply:
You don't need to use the themed mixin with focus-visible since it is a modern feature. You can use CSS variables directly

Reply:
Now using CSS variables within the `focus-visible` mixin.